activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/test/java/org/activiti/cloud/services/modeling/config/GenericModelsConfiguration.java

```diff
@@ -36,219 +40,248 @@
-                             ValidationContext validationContext) {
-            LOGGER.info("validate generic non json content");
-        }
-
-        @Override
-        public ModelType getHandledModelType() {
-            return genericNonJsonModelType;
-        }
-
-        @Override
-        public String getHandledContentType() {
-            return ALL_VALUE;
-        }
-    };
+  private static Logger LOGGER = LoggerFactory.getLogger(GenericModelsConfiguration.class);
 
-    private ModelExtensionsValidator genericNonJsonExtensionsValidator = new ModelExtensionsValidator() {
+  JsonModelType genericJsonModelType = new JsonModelType() {
 
-        @Override
-        public void validate(byte[] modelFile,
-                             ValidationContext validationContext) {
-            LOGGER.info("validate generic non json extensions");
-        }
+    @Override
+    public String getName() {
+      return "GENERIC_JSON_MODEL";
+    }
 
-        @Override
-        public ModelType getHandledModelType() {
-            return genericNonJsonModelType;
-        }
-    };
+    @Override
+    public String getFolderName() {
+      return "generic-json-model";
+    }
+  };
 
-    private ContentUpdateListener genericJsonContentUpdateListener = new ContentUpdateListener() {
+  ModelType genericNonJsonModelType = new ModelType() {
 
-        @Override
-        public ModelType getHandledModelType() {
-            return genericJsonModelType;
-        }
+    @Override
+    public String getName() {
+      return "GENERIC_NON_JSON_MODEL";
+    }
 
-        @Override
-        public void execute(Model model,
-                            FileContent fileContent) {
-            LOGGER.info("generic json content update listener");
-        }
-    };
+    @Override
+    public String getFolderName() {
+      return "generic-non-json-model";
+    }
 
-    private ModelUpdateListener genericJsonModelUpdateListener = new ModelUpdateListener() {
+    @Override
+    public String getContentFileExtension() {
+      return ALL_VALUE;
+    }
+
+    @Override
+    public String[] getAllowedContentFileExtension() {
+      return new String[]{".a", ".b"};
+    }
+  };
 
-        @Override
-        public ModelType getHandledModelType() {
-            return genericJsonModelType;
-        }
+  ModelContentValidator genericJsonContentValidator = new ModelContentValidator() {
 
-        @Override
-        public void execute(Model modelToBeUpdated, Model newModel) {
-            LOGGER.info("generic json model update listener");
-        }
-    };
+    @Override
+    public void validate(byte[] modelFile,
+                         ValidationContext validationContext) {
+      LOGGER.info("validate generic json content");
+    }
 
-    private ContentUpdateListener genericNonJsonContentUpdateListener = new ContentUpdateListener() {
+    @Override
+    public Collection<ModelValidationError> validateAndReturnErrors(byte[] modelContent,
```

Comment:
On beginning assumed that it would be good to keep backward compatibility and just introduce a new method. If this is not a concern then I will just refactor existing method to return errors

Reply:
You're right, it's always good to keep in mind backward compatibility, but in this specific case, those are not public APIs meant to be used by other people, so I think it's fine to refactor it as it better fits for us.